docs/developer/mesh.md

```diff
@@ -0,0 +1,270 @@
+# Mesh
+
+Here we describe the two mesh formarts that are used in RDycore. The two mesh
+formats that we use are:
+
+1. Exodus II
+2. DMPlex HDF5 v3.0.0
+
+RDycore has been tested for meshes that include triangular, quadrilateral, or
+both triangular and quadrilateral cells. The vertices of cells have coordinates
+in 3D.
+
+## Exodus II Mesh Format
+
+The Exodus II mesh format (`.exo`) uses 1-based indices.
+
```

Comment:
I think it's worth explaining here that the mesh consists of cells (triangular or quadrilateral), edges bounding the cells, and vertices at the edge's endpoints. Then we can say that the mesh boundary is represented by one or more "edge sets", and that these "edge sets" identify the edges that make up different parts of the boundary.